src/common/admin_socket.cc

```diff
@@ -274,9 +274,13 @@ void AdminSocket::entry() noexcept
     if (fds[1].revents & POLLIN) {
       // read off one byte
       char buf;
+      #ifndef _WIN32
```

Comment:
Nit: can `safe_recv` just be reused here?

Reply:
We'll have a socket on Windows and a pipe on Linux. I can simply define send_fd as a write_fd wrapper on Linux so that it will work with any type of fd.

At first, I thought that using send_fd with a pipe might seem confusing so I went with explicit platform checks.